### ISSUE TITLE
VS Code: After installation, diag showed on first token

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#718]
https://clips.twitch.tv/SparklingDignifiedCougarResidentSleeper-RHRLrGb3lgZtUhVu

**Comment by strager** [comment]
I couldn't reproduce the issue on VS Code 1.66.0 or 1.67.2 on Linux x86_64.

**Comment by SidneyNemzer** [comment]
I think this is a minimal repro (different error):

https://user-images.githubusercontent.com/21203108/169447519-cff43813-20f7-4492-8315-fadec701e6de.mp4

```
Version: 1.67.2 (system setup)
Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
Date: 2022-05-17T18:15:52.058Z
Electron: 17.4.1
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.19043
```

**Comment by strager** [comment]
I can repro on Windows with VS Code 1.67.2 (x64) using @SidneyNemzer's method.

**Comment by strager** [comment]
I tracked it down to a call to `napi_create_double` creating a strange `napi_value`. More investigation needed.

**Comment by strager** [comment]
The xmm1 register (`double` parameter of `napi_create_double`) gets corrupted by a function called by `__delayLoadHelper2`. [The trampoline generated by GNU binutils' dlltool](https://github.com/bminor/binutils-gdb/blob/d195321cec5d77f48845fc606f8a3a97dc0e0024/binutils/dlltool.c#L572) does not preserve xmm1, despite xmm1 being a volatile register according to the [Windows x64 ABI](https://docs.microsoft.com/en-us/cpp/build/x64-software-conventions?redirectedfrom=MSDN&view=msvc-170).

I think the fix is to save xmm1 and other volatile SSE and AVX registers before calling `__delayLoadHelper2`. I have tried to build binutils, but it looks like the latest version generates archives which crash at run-time. I don't know the exact reasons for the crash yet.

**Comment by strager** [comment]
Patching dlltool fixed the issue. I emailed my tentative patch to upstream.

We still need to apply a workaround in quick-lint-js, since we won't get a newer dlltool for a while.

**Comment by strager** [comment]
Fixed in commit 40d12917dd842571e56d4636053e5573052389a9.
